ntbs-service/Pages/Search/Index.cshtml.cs

```diff
@@ -102,41 +112,118 @@ public async Task<IActionResult> OnGetAsync(int? pageIndex)
+                nextLegacyOffset = null;
+            }
 
-            SetPaginationDetails();
+
+            var notificationBannerModels = notificationsToDisplay.CreateNotificationBanners(User, _authorizationService);
+            SearchResults = new PaginatedList<NotificationBannerModel>(notificationBannerModels, Count, PaginationParameters);
+
+            SetPaginationDetails(nextNtbsOffset, nextLegacyOffset, previousNtbsOffset, previousLegacyOffset, ntbsOffset, legacyOffset);
 
             return Page();
         }
 
+        public async Task<IEnumerable<Notification>> SearchWithoutOffsets(IQueryable<Notification> filteredDrafts, IQueryable<Notification> filteredNonDrafts)
+        {
+            var numberOfNotificationsToFetch = PaginationParameters.PageSize * PaginationParameters.PageIndex;
+            var (orderedNotificationIds, count) = await _searchService.OrderAndPaginateQueryables(filteredDrafts, filteredNonDrafts, PaginationParameters);
+            var notifications = await _notificationRepository.GetNotificationsByIdsAsync(orderedNotificationIds);
+            var (legacyNotifications, legacyCount) = await _legacySearchService.Search(0, numberOfNotificationsToFetch);
+            Count = legacyCount + count;
+            var allPossibleNotifications = notifications.Concat(legacyNotifications);
+            var notificationss = allPossibleNotifications
+                .OrderByDescending(n => n.NotificationStatus == NotificationStatus.Draft)
+                .ThenByDescending(n => n.NotificationDate ?? n.CreationDate)
+                .ThenByDescending(n => n.NotificationId)
+                .ThenBy(n => n.LTBRID ?? n.ETSID)
+                .Skip(numberOfNotificationsToFetch - PaginationParameters.PageSize)
+                .Take(PaginationParameters.PageSize);
+            return notificationss;
+        }
+
+        public async Task<IEnumerable<Notification>> SearchWithOffsets(IQueryable<Notification> filteredDrafts, IQueryable<Notification> filteredNonDrafts)
+        {
+            var (orderedNotificationIds, count) = await _searchService.OrderAndPaginateQueryables(filteredDrafts, filteredNonDrafts, PaginationParameters);
+            var notifications = await _notificationRepository.GetNotificationsByIdsAsync(orderedNotificationIds);
+            var (legacyNotifications, legacyCount) = await _legacySearchService.Search((int)PaginationParameters.LegacyOffset, PaginationParameters.PageSize);
+            Count = legacyCount + count;
+            var allPossibleNotifications = notifications.Concat(legacyNotifications);
+            var notificationss = allPossibleNotifications
+                .OrderByDescending(n => n.NotificationStatus == NotificationStatus.Draft)
+                .ThenByDescending(n => n.NotificationDate ?? n.CreationDate)
+                .ThenByDescending(n => n.NotificationId)
+                .ThenBy(n => n.LTBRID ?? n.ETSID)
+                .Take(PaginationParameters.PageSize);

```

Comment:
This repeats with above a bit. I agree that two functions, rather than a `shouldUseOffsets` boolean is better, but it'd still be good to commonize it more perhaps?